applications/tari_base_node/src/recovery.rs

```diff
@@ -0,0 +1,166 @@
+    // create recovery db
+    match &node_config.db_type {
+        DatabaseType::LMDB(p) => {
+            let _backend = create_recovery_lmdb_database(&p).map_err(|err| {
+                error!(target: LOG_TARGET, "{}", err);
+                ExitCodes::UnknownError
+            })?;
+        },
+        _ => {
+            error!(target: LOG_TARGET, "Recovery mode is only available for LMDB");
+            return Err(ExitCodes::UnknownError);
+        },
+    };
+    Ok(())
+}
+
+pub async fn run_recovery(node_config: &GlobalConfig) -> Result<(), String> {
+    println!("string recovery mode");
+    let (temp_db, main_db) = match &node_config.db_type {
+        DatabaseType::LMDB(p) => {
+            let backend =
+                create_lmdb_database(&p, node_config.db_config.clone(), MmrCacheConfig::default()).map_err(|e| {
+                    error!(target: LOG_TARGET, "Error opening db: {}", e);
+                    format!("Could not open DB: {}", e.to_string())
+                })?;
+            let new_path = Path::new(&p).join("temp_recovery");
+
+            let temp = create_lmdb_database(&new_path, node_config.db_config.clone(), MmrCacheConfig::default())
+                .map_err(|e| {
+                    error!(target: LOG_TARGET, "Error opening recovery db: {}", e);
+                    format!("Could not open recovery DB: {}", e.to_string())
+                })?;
+            (temp, backend)
+        },
+        _ => {
+            error!(target: LOG_TARGET, "Recovery mode is only available for LMDB");
+            Err(format!("Recovery mode is only available for LMDB"))?
+        },
+    };
+    let rules = ConsensusManagerBuilder::new(node_config.network.into()).build();
+    let factories = CryptoFactories::default();
+    let validators = Validators::new(
+        FullConsensusValidator::new(rules.clone()),
+        StatelessBlockValidator::new(rules.clone(), factories.clone()),
+    );
+    let db_config = BlockchainDatabaseConfig {
+        orphan_storage_capacity: node_config.orphan_storage_capacity,
+        pruning_horizon: node_config.pruning_horizon,
+        pruning_interval: node_config.pruned_mode_cleanup_interval,
+    };
+    let db = BlockchainDatabase::new(main_db, &rules, validators, db_config).map_err(|e| {
+        error!(target: LOG_TARGET, "Error could not create BlockchainDatabase {}", e);
+        format!("Could not create BlockchainDatabase: {}", e.to_string())
+    })?;
+    do_recovery(db, temp_db).await?;
+
+    info!(
+        target: LOG_TARGET,
+        "Node has completed recovery mode, it will try to cleanup the db"
+    );
+    match &node_config.db_type {
+        DatabaseType::LMDB(p) => {
+            let new_path = Path::new(p).join("temp_recovery");
+            remove_lmdb_database(&new_path).map_err(|e| {
+                error!(target: LOG_TARGET, "Error removing recovery db: {}", e);
+                format!("Could not remove recovery DB: {}", e.to_string())
+            })
```

Comment:
We log any errors anyway - so your code could be cleaned up in a number of places by just using `?` and `anyhow`.

```suggestion
            remove_lmdb_database(&new_path)?;
```